torchtune/data/img_gen_data.rfc.md

```diff
@@ -0,0 +1,162 @@
+# Overview
+
+This is an RFC regarding how we should support datasets for finetuning text-conditioned image generation models.
+
+A basic data pipeline for this would be:
+1. Load the JSON/CSV/TSV/Parquet/LMDB/etc. file containing the image paths/urls and captions
+2. For each pair:
+   - load/download the image
+   - resize the image and optionally randomly augment it (horizontal flip, etc.) and normalize it
+   - optionally randomly augment the caption (rearrange caption parts, etc.)
+   - tokenize the caption using the model's tokenizer
+3. collate into a batch
+
+At a broad level, this fits well into our current TorchTune data ecosystem (except we wouldn't use the "list of Message objects" abstraction, which would change how we interact with the model's tokenizer).
+
+In TorchTune, a simple version would look something like this:
+
+```yaml
+dataset:
+    _component_: torchtune.datasets.img_caption_dataset
```

Comment:
Maybe just my naivate, but when I hear image-caption dataset, I assume it's a dataset for taking an image and generating a caption, which is not the case here. 

Hugging Face has a label for these datasets called ["Text-to-Image"](https://huggingface.co/datasets?task_categories=task_categories%3Atext-to-image), which I think is a more accurate description. This also is inline with our addition of task-centered dataset builders like the [vqa_dataset](torchtune/datasets/multimodal/_vqa.py).

Concretely proposing changing the default dataset for diffusion from `img_caption_dataset` to `text_to_image_dataset`

Reply:
I figured that this dataset could be used for any downstream task that uses pairs of images+text. Like finetuning CLIP for example. Maybe `image_text_pair_dataset`? Or is it more clear for the user if we name the datasets based on a specific use of it?